app/src/main/java/org/listenbrainz/android/util/JobQueue.kt

```diff
@@ -0,0 +1,169 @@
+package org.listenbrainz.android.util
+
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.CoroutineExceptionHandler
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.CoroutineStart
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.cancelAndJoin
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.ensureActive
+import kotlinx.coroutines.flow.receiveAsFlow
+import kotlinx.coroutines.isActive
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+import java.util.LinkedList
+import java.util.concurrent.atomic.AtomicBoolean
+import kotlin.coroutines.CoroutineContext
+
+class JobQueue(
+    private val dispatcher: CoroutineDispatcher,
+    private val log: Log = Log.Companion
+) {
+    private val exceptionHandler = CoroutineExceptionHandler { _, e ->
+        log.e("JobQueue Exception: ${e.message}")
+    }
+    private val scope = CoroutineScope(SupervisorJob() + dispatcher)
```

Comment:
Add the exceptionHandler as well here

Reply:
We don't want to handle exceptions in any other coroutines. I want the app to crash so we can identify any malfunctions (Since we don't have QA). 99% of the time, the there will be no effect.

Reply:
This is a really bad practice @07jasjeet . We cannot afford to crash the app in any situation. Tbh even if there is an error, don't crash. Rather send toasts or errors. A crash leads to a direct 1 star rating of the app. Also, facing a crash helps no one. It just is a very bad experience. Move this.

Reply:
Well I would still want the current behaviour in beta but if you say so, Ill add the handler.